### ISSUE TITLE
Hover color transition does not work with incremental layout

### ISSUE THREAD

**Issue by jdm** [servo/servo#17667]
```html
<style>
div { transition: color .17s; color: black }
div:hover { color: red; }
</style>
<div>hi there</div>
```
When run with `-i`, the color changes to red (with no intervening interpolated colors). Without `-i`, the text never changes color.

**Comment by moonlightdrive** [comment]
Hi @jdm I'd like to take a look at this one. Do you have any advice on where to start?

**Comment by mbrubeck** [comment]
For bugs in incremental layout, my first step is to check that the correct "restyle damage" is inserted when an element's style changes.  For CSS transitions, the style change happens in [update_style_for_animation][1].  I notice this method's doc comment talks about a `damage` parameter that no longer exists since https://github.com/servo/servo/commit/6d67525172811f203220f02defa7b4635e6414c8, which makes me suspicious that this is a regression from that commit.

It looks like it's now up to the caller of `update_style_for_animation` to insert the correct damage.  The function is called in two places: [layout::animation::recalc_style_for_animations][2] and [style::matching::process_animations][3].  Using a debugger or adding some logging to these functions should help determine which is getting called, and what restyle damage they are inserting.

The restyle damage is calculated based on the differences between "old" and "new" styles (i.e. the computed values before and after the change is applied).  Glancing at [this code][4], I'm suspicious that this is taking a snapshot of the "old" values *after* the animation has been applied to them.

[1]: https://github.com/servo/servo/blob/c2deac171d2407579b4fd0284b5f3c71be696f2d/components/style/animation.rs#L611-L613
[2]: https://github.com/servo/servo/blob/c2deac171d2407579b4fd0284b5f3c71be696f2d/components/layout/animation.rs#L158-L168
[3]: https://github.com/servo/servo/blob/c2deac171d2407579b4fd0284b5f3c71be696f2d/components/style/matching.rs#L303
[4]: https://github.com/servo/servo/blob/c2deac171d2407579b4fd0284b5f3c71be696f2d/components/style/matching.rs#L500-L509

**Comment by mrobinson** [comment]
This was fixed by #26407. I believe the reason this was failing was that the fragment that contained the text was inheriting the style from its containing box. The code that updated the style for transitions in the layout thread didn't descend into children when updating transitioning inherited properties. Now that animations update style during restyle, inheritance should work properly.
